cmd/thanos/compact.go

```diff
@@ -436,6 +436,12 @@ func runCompact(
 		level.Info(logger).Log("msg", "retention policy of 1 hour aggregated samples is enabled", "duration", retentionByResolution[compact.ResolutionLevel1h])
 	}
 
+	retentionByTenant, err := compact.ParesRetentionPolicyByTenant(logger, *conf.retentionTenants)
+	if err != nil && len(retentionByTenant) != 0 {
```

Comment:
Why is `len(retentionByTenant) != 0` needed here? If it's an empty config, `err` should already be nil.

Reply:
good catch